### PR TITLE
[RI Ch05] Installer Requirements - issue 846

### PR DESCRIPTION
**Pull request by ChilionChan**
Hi, adding a new PR for issue #846#  , @fuqiao123 @mf4716 

contents: 
a) adding missing hardware attributes 
b) adding generic information (Metadata Lab Owner, Location, Purpose etc..) . 
c) enhance NIC Mapping with server instances. 
d) embedding a descriptor file schema file.
f) adding appendix for sample description file

### CODE REVIEW

File: doc/ref_impl/cntt-ri/chapters/chapter05.md [hunk 1]
```diff
@@ -127,16 +136,11 @@ Besides it may include additional information pim username, password, rack_name,
 | device_name | String | Yes | e.g NFV-D-HDBNJ-02A-3503-G-02-M-SRV-01 |
 | system_diskname | String | No | system disk name |
 | system_disksize | String | No | system disk size |
-| pim_ip_address | String | Yes | PIM ip address |
-| pim_netmask | String | Yes | PIM netmask |
-| pim_username | String | Yes | PIM user |
-| pim_password | String | Yes | PIM password |
+| role | String | No | computer or controller |
```
[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/ref_impl/cntt-ri/chapters/chapter05.md]: @markbeierl in "... we can have dedicated storage nodes." the operative word is "can" viz. operators to decide.

[the remark this answers is not in the document]
Comment (markbeierl) [reply doc/ref_impl/cntt-ri/chapters/chapter05.md]: Correct, but what does the role mean in this case?  There is no mention of storage.

File: doc/ref_impl/cntt-ri/chapters/chapter05.md [hunk 2]
```diff
@@ -90,10 +97,12 @@ Server template would be assigned to multiple servers, i.e physical hosts.
...
 | memory | String | Yes |  |
-| hard_drive | String | Yes |  |
+| disk | String | Yes |  |
 | raid | String | Yes |  |
-| network_card_infos | List | No | interface list definition|
-| network_card_bond_infos | List | No | NIC bonding, might not be always the case.|
+| interfaces | List | No | interface list definition|
+| bond_interfaces | List | No | NIC bonding, might not be always the case.|
+| remote_user | String | Yes | remote user, all server setting same remote user |
+| remote_password | String | Yes | remote password, all server setting same remote password |
```
[the remark this answers is not in the document]
Comment (ChilionChan) [reply doc/ref_impl/cntt-ri/chapters/chapter05.md]: Right. IPMI,BMC,ILO etc, all servers with same credential.

File: doc/ref_impl/cntt-ri/chapters/chapter05.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChilionChan) [reply doc/ref_impl/cntt-ri/chapters/chapter05.md]: I'll add storage in the enum type list.

**Comment by rabi-abdel** [comment]
@fuqiao123 @ChilionChan due to file names changes, it will be probably better to close this PR and start a new one (also it is possible the content of xDF can be added to Chapter 9 under development.

let me know what you think.

**Comment by fuqiao123** [comment]
> @fuqiao123 @ChilionChan due to file names changes, it will be probably better to close this PR and start a new one (also it is possible the content of xDF can be added to Chapter 9 under development.
> 
> let me know what you think.

Hi, Rabi. I am OK with merging this PR. However I don't think chapter 9 is suitable for xDF. xDF is something we are pushing the installers and testing tools to follow I think for the following releases. It is a basic requirement rather than just a gap or innovation. I would rather create a new chapter for the xDF. In the meantime, xDF is in Yaml format, and is still developing. Per our discussion previously, it is still not clear whether we should put that in the CIRV repo for OPNFV or put it here in the doc. I would suggest for now we just make it as it is, put it as an appendix for chp 6. And we will revisit this once the discussion of xDF is more clear within the community. What do you say?

**Comment by rabiabdel** [comment]
> > @fuqiao123 @ChilionChan due to file names changes, it will be probably better to close this PR and start a new one (also it is possible the content of xDF can be added to Chapter 9 under development.
> > let me know what you think.
> 
> Hi, Rabi. I am OK with merging this PR. However I don't think chapter 9 is suitable for xDF. xDF is something we are pushing the installers and testing tools to follow I think for the following releases. It is a basic requirement rather than just a gap or innovation. I would rather create a new chapter for the xDF. In the meantime, xDF is in Yaml format, and is still developing. Per our discussion previously, it is still not clear whether we should put that in the CIRV repo for OPNFV or put it here in the doc. I would suggest for now we just make it as it is, put it as an appendix for chp 6. And we will revisit this once the discussion of xDF is more clear within the community. What do you say?

Hi @fuqiao123 yes this sounds good. Please approve this PR as it requires your approval before it can be merged.

**Comment by lylavoie** [comment]
Recommend removing the sample schema file, as this is still being worked as part of the HDV projects and other efforts, and the specific yaml file will likely change in that work.

**Comment by ChilionChan** [comment]
I remove the sample desription file from the content, thanks to review it again.

**Comment by mf4716** [comment]
@markbeierl , @pgoyal01 , @deebhatia , @mkr1481 , @collivier , @lylavoie , @rrajesh70 -

Reviewers, please review for approval at your next opportunity.

The PR is a little old, still relevant, and we'd like to merge it this week with your approvals.

Thanks in advance.

**Comment by pgoyal01** [comment]
5.3 can benefit from a definition of the entities and their relationships to each other, and a schema definition. The OPNFV PDF has issues but it is a good starting point -- they lack an entity view. 

As an example of what is required, A compute node (in RI -- server; BTW in OSTK a server is logical and refers to VM/container) is composed of other entities CPU sockets, NIC slots, attached disks, etc. For each of these there is a number (such as 2 CPU sockets, 2 NIC slots, ...). These entities themselves are compositions: NIC 2 or 4 ports, each port has a speed, etc.

The descriptor file can then be organised according to the entities and their components, and where the components are references to other entities.

**Comment by pgoyal01** [comment]
Sorry -- closed mistakenly.

**Pull request merged**